### ISSUE TITLE
Pods admin user interface forces 0 for empty numeric fields

### ISSUE THREAD

**Issue by tonnt** [pods-framework/pods#6226]
## Description

When having a numeric field (plain number) for a pod, the user interface automatically replaces all empty values with 0 when adding new data or editing existing data. It is not possible to delete the automatic zeros because javascript automatically replaces the empty field again with 0.

## Testing instructions

4. See error message '....'

## Expected behavior

Empty fields should stay empty not being automatically populated with 0.

## Screenshots / screencast

![image](https://user-images.githubusercontent.com/3164213/138568002-a5f90784-7e33-4cbf-bf8c-a1c6d52ecedc.png)

## Pods version

2.8.1

## Other super helpful information

<details>
<summary>WordPress Environment</summary>
`
### wp-core ###

version: 5.8.1
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: /%postname%/
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: undefined
environment_type: production
user_count: 123
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /home/customer/www/fundamentaltrend.com/public_html
wordpress_size: 44.22 MB (46372511 bytes)
uploads_path: /home/customer/www/fundamentaltrend.com/public_html/wp-content/uploads
uploads_size: 74.83 MB (78467835 bytes)
themes_path: /home/customer/www/fundamentaltrend.com/public_html/wp-content/themes
themes_size: 5.46 MB (5726233 bytes)
plugins_path: /home/customer/www/fundamentaltrend.com/public_html/wp-content/plugins
plugins_size: 151.83 MB (159210009 bytes)
database_size: 126.92 MB (133087232 bytes)
total_size: 403.27 MB (422863820 bytes)

### wp-active-theme ###

name: Hello Elementor Child (hello-elementor-child)
version: 1.0.0
author: TT
author_website: (undefined)
parent_theme: Hello Elementor (hello-elementor)
theme_features: core-block-patterns, widgets-block-editor, post-thumbnails, menus, automatic-feed-links, title-tag, html5, custom-logo, editor-style, align-wide, woocommerce, wc-product-gallery-zoom, wc-product-gallery-lightbox, wc-product-gallery-slider
theme_path: /home/customer/www/fundamentaltrend.com/public_html/wp-content/themes/hello-elementor-child
auto_update: Disabled

### wp-parent-theme ###

name: Hello Elementor (hello-elementor)
version: 2.4.1
author: Elementor Team
author_website: https://elementor.com/?utm_source=wp-themes&utm_campaign=author-uri&utm_medium=wp-dash
theme_path: /home/customer/www/fundamentaltrend.com/public_html/wp-content/themes/hello-elementor
auto_update: Disabled

### wp-themes-inactive (2) ###

Twenty Nineteen: version: 2.1, author: the WordPress team, Auto-updates disabled
Twenty Twenty-One: version: 1.4, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (20) ###

Elementor: version: 3.4.6, author: Elementor.com, Auto-updates enabled
Elementor Pro: version: 3.4.2, author: Elementor.com, Auto-updates disabled
GeoTargetingWP: version: 3.3.8.1, author: Timersys, Auto-updates disabled
If Menu - Visibility control for menu items: version: 0.16.2, author: Layered, Auto-updates disabled
JetBlocks For Elementor: version: 1.2.10, author: Crocoblock, Auto-updates disabled
JetBlog For Elementor: version: 2.2.14, author: Crocoblock, Auto-updates enabled
JetElements For Elementor: version: 2.5.9, author: Crocoblock, Auto-updates disabled
MakeCommerce: version: 3.0.8, author: Maksekeskus AS, Auto-updates disabled
MemberPress Elementor Content Protection: version: 1.0.2, author: Caseproof, LLC, Auto-updates disabled
MemberPress Plus 10 (Legacy): version: 1.9.24, author: Caseproof, LLC, Auto-updates disabled
Memberpress WooCommerce Plus: version: 1.1.4, author: HappyPlugins, Auto-updates disabled
Min and Max Quantity for WooCommerce: version: 1.3.2.1, author: BeRocket, Auto-updates disabled
Pods - Custom Content Types and Fields: version: 2.8.1, author: Pods Framework Team, Auto-updates enabled
Site Kit by Google: version: 1.43.0, author: Google, Auto-updates disabled
User Role Editor Pro: version: 4.60.2, author: Vladimir Garagulia, Auto-updates disabled
WooCommerce: version: 5.8.0, author: Automattic, Auto-updates disabled
WooCommerce PDF Invoices & Packing Slips: version: 2.10.1, author: Ewout Fernhout, Auto-updates disabled
WooCommerce Price Based on Country (Basic): version: 2.0.27, author: Oscar Gare, Auto-updates disabled
WooCommerce Price Based on Country Pro Add-on: version: 2.11.3, author: Oscar Gare, Auto-updates disabled
WooCommerce Subscriptions: version: 3.0.12, author: WooCommerce, Auto-updates disabled

### wp-plugins-inactive (3) ###

EveryPay payment gateway for WooCommerce: version: 1.3.4, author: EveryPay AS, Auto-updates disabled
SiteGround Optimizer: version: 5.9.7, author: SiteGround, Auto-updates disabled
User Role Editor: version: 4.60.2, author: Vladimir Garagulya, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_GD
imagick_module_version: Not available
imagemagick_version: Not available
imagick_version: Not available
file_uploads: File uploads is turned off
post_max_size: 256M
upload_max_filesize: 256M
max_effective_size: 256 MB
max_file_uploads: 20
gd_version: 2.2.5
gd_formats: GIF, JPEG, PNG, WebP, BMP, XPM
ghostscript_version: 9.27

### wp-server ###

server_architecture: Linux 3.12.18-clouder0 x86_64
httpd_software: Apache
php_version: 7.4.25 64bit
php_sapi: cgi-fcgi
max_input_variables: 3000
time_limit: 120
memory_limit: 768M
max_input_time: 120
upload_max_filesize: 256M
php_post_max_size: 256M
curl_version: 7.66.0 OpenSSL/1.1.1l-fips
suhosin: false
imagick_availability: false
pretty_permalinks: true
htaccess_extra_rules: true

### wp-database ###

extension: mysqli
server_version: 5.7.32-35-log
client_version: mysqlnd 7.4.25

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /home/customer/www/fundamentaltrend.com/public_html/wp-content
WP_PLUGIN_DIR: /home/customer/www/fundamentaltrend.com/public_html/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 768M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_LOCAL_DEV: undefined
DB_CHARSET: utf8mb4
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

### google-site-kit ###

version: 1.43.0
php_version: 7.4.25
wp_version: 5.8.1
reference_url: https://fundamentaltrend.com
amp_mode: no
site_status: connected-site
user_status: authenticated
verification_status: verified-non-site-kit
connected_user_count: 2
active_modules: site-verification, search-console, analytics, analytics-4
required_scopes: 
	openid: ✅
	https://www.googleapis.com/auth/userinfo.profile: ✅
	https://www.googleapis.com/auth/userinfo.email: ✅
	https://www.googleapis.com/auth/siteverification: ✅
	https://www.googleapis.com/auth/webmasters: ✅
	https://www.googleapis.com/auth/analytics.readonly: ✅
capabilities: 
	googlesitekit_authenticate: ✅
	googlesitekit_setup: ✅
	googlesitekit_view_posts_insights: ✅
	googlesitekit_view_dashboard: ✅
	googlesitekit_view_module_details: ✅
	googlesitekit_manage_options: ✅
enabled_features: 
	ga4setup: ✅
	ideaHubModule: ⭕
	serviceSetupV2: ⭕
	swgModule: ⭕
	userInput: ⭕
	unifiedDashboard: ⭕
search_console_property: https://fundamentaltrend.com/
analytics_account_id: 1879•••••
analytics_property_id: UA-1879•••••••
analytics_profile_id: 2361•••••
analytics_use_snippet: yes
analytics_4_property_id: none
analytics_4_web_data_stream_id: none
analytics_4_measurement_id: none
analytics_4_use_snippet: yes

### pods ###

pods-server-software: Apache
pods-user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36
pods-session-save-path: /tmp
pods-session-save-path-exists: Yes
pods-session-save-path-writable: Yes
pods-session-max-lifetime: 1440
pods-opcode-cache-apc: No
pods-opcode-cache-memcached: No
pods-opcode-cache-opcache: Yes
pods-opcode-cache-redis: No
pods-object-cache-apc: No
pods-object-cache-apcu: No
pods-object-cache-memcache: No
pods-object-cache-memcached: Yes
pods-object-cache-redis: No
pods-memory-current-usage: 102.845M
pods-memory-current-usage-real: 104.000M
pods-network-wide: No
pods-install-location: /home/customer/www/fundamentaltrend.com/public_html/wp-content/plugins/pods/
pods-developer: No
pods-tableless-mode: No
pods-relationship-table-enabled: Yes
pods-light-mode: No
pods-strict: No
pods-allow-deprecated: Yes
pods-api-cache: Yes
pods-shortcode-allow-evaluate-tags: No
pods-sessions: Disable sessions
pods-can-use-sessions: Yes

`  
```
Copy and paste your Tools > Site Health > Info. Use the 'Copy Site Info to Clipboard' and paste the results over this line.
```

</details>

<details>
<summary>Pods Package Export</summary>

```json
Copy and paste your Pods Admin > Migrate: Packages > Export and paste the code over this line.
```

</details>

**Comment by sc0ttkclark** [comment]
Fixed in 2.8.2

**Comment by sc0ttkclark** [comment]
Duplicate of #6209
